src_common/common_linux.inl

```diff
@@ -1283,6 +1285,57 @@ void terminate_process_without_waiting(Process* process)
 #define SYS_pidfd_open 434
 #endif
 
+bool wait_for_process(struct Process* process, double seconds, u32* out_exit_code)
+{
+    if (process->pid == 0)
+    {
+        *out_exit_code = -1;
+        return false;
+    }
+
+    pid_t pid = process->pid;
+
+    if (seconds == WAIT_FOR_PROCESS_FOREVER)
+    {
+        int status;
+        if (waitpid(pid, &status, 0) > 0)
+        {
+            if (WIFEXITED(status))
+                *out_exit_code = WEXITSTATUS(status);
+            else
+                *out_exit_code = -1;
+
+            process->pid = 0;
+            return true;
+        }
+    }
+    else
+    {
+        int status;
+        QPC qpc_timeout = qpc_from_seconds(seconds);
+        QPC start_time  = current_qpc();
+        do
+        {
+            if (waitpid(pid, &status, WNOHANG) > 0)
```

Comment:
Check for EINTR, in case the call was interrupted by a signal handler. https://linux.die.net/man/7/signal

Reply:
Nevermind, seems WNOHANG guarantees EINTR is impossible.